05_stack_and_queue/src/stack.c

```diff
@@ -0,0 +1,64 @@
+#include "stack.h"
+#include <stdbool.h>
+#include <stdlib.h>
+
+typedef struct StackNode {
+    char data;
+    struct StackNode* next;
+} StackNode;
+
+typedef struct Stack {
+    StackNode* head;
+} Stack;
+
+Stack* newStack(void)
+{
+    Stack* stack = calloc(1, sizeof(Stack));
+    return stack;
+}
+
+bool isEmpty(Stack* stack)
+{
+    return stack->head == NULL;
+}
+
+void push(Stack* stack, char data)
+{
+    StackNode* element = (StackNode*)calloc(1, sizeof(StackNode));
+
+    element->data = data;
+    element->next = stack->head;
+    stack->head = element;
+}
+
+char pop(Stack* stack)
+{
+    if (isEmpty(stack)) {
+        return '\0';
+    }
+
+    StackNode* oldNode = stack->head;
+    char data = oldNode->data;
+    stack->head = stack->head->next;
+
+    free(oldNode);
+
+    return data;
+}
+
+char peek(Stack* stack)
+{
+    if (isEmpty(stack)) {
+        return '\0';
+    }
+
+    return stack->head->data;
+}
+
+void deleteStack(Stack* stack)
+{
+    while (!isEmpty(stack)) {
+        pop(stack);
+    }
+    free(stack);
+}
```

Comment:
Nit: обычно функции удаления хорошо делать устойчивыми к `NULL` входам.
```suggestion
void deleteStack(Stack* stack)
{
    if (stack == NULL) return;

    while (!isEmpty(stack)) {
        pop(stack);
    }
    free(stack);
}
```